src/core/libmaven/Compound.h

```diff
@@ -65,12 +71,52 @@ class Compound{
         string id;          /**@param - compund id */
 
         string name;        /**@param -  name of compound */
-        string formula;         /**@param -  formula of compound */
         string kegg_id;         /**@param -  kegg_id-    Kyoto Encyclopedia of Genes and Genomes id*/
         string pubchem_id;      /**@param  -  pubchem_id -    PubChem id*/
         string hmdb_id;         /**@param  -  hmdb_id-    Human Metabolome Database id */
         string alias;       /**@param   -  alias name of compound   */
 
+        /**
+         * @brief Obtain the formula set for this compound.
+         * @return A string representing a valid chemical formula.
+         */
+        string formula() const { return _formula; }
+
+        /**
+         * @brief Set the formula of this compound.
+         * @details Some basic filtering will be done on the formula string
+         * provided as the argument.
+         * @param formula Ideally, an alpha-numeric string, representing a valid
+         * chemical formula.
+         */
+        void setFormula(string formula) { _formula = filterFormula(formula); }
+
+        /**
+         * @brief Filters an arbitrary string and gets rid of any characters
+         * that are not allowed to be part of a small molecule formula.
+         * @details It should be noted that this is only a filter and not a
+         * parser of any sort. This function can be used to strip a possible
+         * formula string to a form which is equivalent to how El-MAVEN
+         * currently understands a checmical formula. The parsing scheme in
+         * `MassCalculator::getComposition` can be used as a guide.
+         * @param formulaString Any arbitrary string.
+         * @return A string stripped of all non-chemical characters.
+         */
+        static string filterFormula(string formulaString)
+        {
```

Comment:
@saifulbkhan How about stripping extra spaces within the same function? since that leads to mass calculation errors within El-MAVEN as well. Or do you want to take that up separately?

Reply:
@shubhra-agrawal This function will remove any character from the `formulaString` that is not part of the combined string `CHE_FORMULA_ALPHA_UPP + CHE_FORMULA_ALPHA_LOW + CHE_FORMULA_COFF`. That includes spaces too, so, I am guessing we do not need to filter for them separately.

Reply:
You're right. Resolving this.